src/controllers/controller.doughnut.js

```diff
@@ -19,25 +19,25 @@ defaults._set('doughnut', {
 		mode: 'single'
 	},
 	legendCallback: function(chart) {
-		var text = [];
-		text.push('<ul class="' + chart.id + '-legend">');
-
+		var list = document.createElement('ul');
 		var data = chart.data;
 		var datasets = data.datasets;
 		var labels = data.labels;
 
-		if (datasets.length) {
-			for (var i = 0; i < datasets[0].data.length; ++i) {
-				text.push('<li><span style="background-color:' + datasets[0].backgroundColor[i] + '"></span>');
-				if (labels[i]) {
-					text.push(labels[i]);
-				}
-				text.push('</li>');
+		var i, ilen, listItem, listItemSpan;
+
+		list.setAttribute('class', chart.id + '-legend');
+
+		for (i = 0, ilen = datasets[0].data.length || 0; i < ilen; ++i) {
```

Comment:
Shouldn't we check `datasets.length` before this line?

Reply:
`|| 0` is not needed in that case

Reply:
Hi @nagix , this change was made in response to a request that @benmccann made here:
https://github.com/ian-pvd/Chart.js/pull/1#discussion_r272262833

Reply:
The request I made was specifically referring to `plugin.legend.js` which doesn't check `datasets.length` today because it doesn't access `datasets[0]: https://github.com/chartjs/Chart.js/blob/ddee91eb9f7f68d64b1634c103ebbc8421b23c7a/src/plugins/plugin.legend.js#L76

I would agree that we should not remove the `if (datasets.length)` check here. Good catch on that @nagix

Reply:
@ian-pvd just a quick reminder that if you're able to address the comments from nagix that we'd love to get this PR merged

Reply:
Hey @benmccann , I'd love to, I just haven't had a ton of time this week. I can make it a priority today and get changes pushed up by say.. 8pm EST tonight?

Reply:
Sure. No worries